### PR TITLE
CX2 - MPI benchmarking - WIP

### PR DESCRIPTION
**Pull request by georgebisbas**
Current testing scripts for performance evaluation on cx2.
Work in progress

TODO:
compiler -> ICC
Separate script for installation and dependencies.
Add a README
Clean-up duplication etc...

edit
folder name: mpi-manuscript

### CODE REVIEW

File: mpi-test-scripts/mpi_devito.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+## Mail notification
+#PBS -m bea
+#PBS -M g.bisbas18@imperial.ac.uk
+
+echo Working Directory is $PBS_O_WORKDIR
+cd $PBS_O_WORKDIR
+rm -f stdout* stderr*
+conda remove --name devito --all
+
+# Load gcc and mpi/intel-2018
```
Comment (FabioLuporini) [inline mpi-test-scripts/mpi_devito.sh]: useless line

File: mpi-test-scripts/mpi_devito.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+# Install devito
+module load gcc/8.2.0
+module load intel-suite/2017
```
Comment (FabioLuporini) [inline mpi-test-scripts/mpi_devito.sh]: no 2017?

File: mpi-test-scripts/mpi_devito.sh [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+cd ../
+export DEVITO_HOME=`pwd`/devito
+cd devito/benchmarks/user/
+
+
+# Start time
+echo Start time is `date` > date
+
+# Set env variables
+export DEVITO_ARCH=gcc
```
Comment (FabioLuporini) [inline mpi-test-scripts/mpi_devito.sh]: forget about gcc , use icc

File: mpi-paper/install_devito_full.sh [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+cd devito
+
+# Load anaconda3
+module load anaconda3/personal
+
+# Create devito environment
+conda env create -f environment.yml
+source activate devito
+pip install -e .
+#conda install gcc_linux-64
```
Comment (FabioLuporini) [inline mpi-paper/install_devito_full.sh]: drop

File: mpi-paper/install_devito_full.sh [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+echo ...DONE
+
+mkdir opesci
+cd opesci
+## Installing devito
+git clone https://github.com/opesci/devito.git
+cd devito
+
+# Load anaconda3
+module load anaconda3/personal
```
Comment (FabioLuporini) [inline mpi-paper/install_devito_full.sh:21]: forgot anaconda-setup

File: mpi-paper/install_devito_full.sh [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+# Create devito environment
+conda env create -f environment.yml
+source activate devito
+pip install -e .
+#conda install gcc_linux-64
+conda install mpi4py
```
Comment (FabioLuporini) [inline mpi-paper/install_devito_full.sh]: pip install mpi4py

works

File: mpi-paper/install_devito_full.sh [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+# Install dependencies (optional as well for MPI)
+pip install -r requirements-optional.txt
```
Comment (FabioLuporini) [inline mpi-paper/install_devito_full.sh]: drop

File: mpi-paper/install_devito_full.sh [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+cd $HOME/opesci
+
+# Install OpesciBench
+git clone https://github.com/opesci/opescibench.git
+cd opescibench
+git status
```
Comment (FabioLuporini) [inline mpi-paper/install_devito_full.sh]: drop

File: mpi-paper/install_devito_full.sh [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+git checkout mpi-perf-fixes
+
```
Comment (FabioLuporini) [inline mpi-paper/install_devito_full.sh:34]: drop

File: mpi-paper/install_devito_full.sh [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+# Installing OpesciBench
```
Comment (FabioLuporini) [inline mpi-paper/install_devito_full.sh]: drop

File: mpi-paper/install_devito_full.sh [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+pip install -e .
+# Navigate to benchmarks folder
```
Comment (FabioLuporini) [inline mpi-paper/install_devito_full.sh]: drop

File: mpi-paper/demo_submit_devito_mpi_job.sh [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+mpicc -v
+echo ...DONE
+
+
+source activate devito
+#conda install mpi4py
+python print_mpi4py_conf.py
+
+cd $HOME/opesci/
+module load anaconda3/personal
```
Comment (FabioLuporini) [inline mpi-paper/demo_submit_devito_mpi_job.sh:28]: redundant

Comment (georgebisbas) [reply mpi-paper/demo_submit_devito_mpi_job.sh:28]: fixed

File: mpi-paper/install_devito_full.sh (hunk 11 above)
Comment (georgebisbas) [reply mpi-paper/install_devito_full.sh]: dropped

File: mpi-paper/install_devito_full.sh (hunk 10 above)
Comment (georgebisbas) [reply mpi-paper/install_devito_full.sh]: dropped

File: mpi-paper/install_devito_full.sh (hunk 9 above)
Comment (georgebisbas) [reply mpi-paper/install_devito_full.sh:34]: dropped

File: mpi-paper/install_devito_full.sh (hunk 8 above)
Comment (georgebisbas) [reply mpi-paper/install_devito_full.sh]: dropped

File: mpi-paper/install_devito_full.sh (hunk 7 above)
Comment (georgebisbas) [reply mpi-paper/install_devito_full.sh]: dropped

File: mpi-paper/install_devito_full.sh (hunk 6 above)
Comment (georgebisbas) [reply mpi-paper/install_devito_full.sh]: ok

File: mpi-paper/install_devito_full.sh (hunk 4 above)
Comment (georgebisbas) [reply mpi-paper/install_devito_full.sh]: dropped

File: mpi-paper/install_devito_full.sh (hunk 5 above)
Comment (georgebisbas) [reply mpi-paper/install_devito_full.sh:21]: One should only do it once, maybe add as a directive somwehere else?

**Comment by georgebisbas** [comment]
@FabioLuporini do we need this anymore?

**Comment by FabioLuporini** [comment]
mmmmm maybe not what do you think?

**Comment by georgebisbas** [comment]
> mmmmm maybe not what do you think?

I think from the time make-pbs.py came into play this is useless.

**Pull request closed without merging**